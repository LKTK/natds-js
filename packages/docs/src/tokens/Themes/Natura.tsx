import * as React from 'react';
import { themes } from '@naturacosmeticos/natds-styles';
import Table from '@addons/Table/Table';

const { natura } = themes;

function FlattenTheme(accum: any, [key, value]: any): any {
  const type = typeof value;

  if (type === 'function' || type === 'object' && !!value) {
    return Object.entries(value).reduce(FlattenTheme, [key, ...accum]);
  }

  const [parent] = accum;

  if (!parent) return accum;

  const name = `${parent}-${key}`;

  return [...accum, [{ name }, { name: value, type: 'color' }]];
}

function removeStrings(entry: any) {
  return typeof entry !== 'string';
}

const { light, dark } = natura;

const darkSchema = {
  head: ['name', 'color'],
  body: (dark && dark.palette) && Object.entries(dark.palette).reduce(FlattenTheme, []).filter(removeStrings)
};

darkSchema.body.push(
  [
    {
      name: 'webFontFamilyBranding',
      type: 'text'
    },
    {
      name: dark.typography.web.fontFamilyBrand,
      type: 'text'
    }
  ],
  [
    {
      name: 'androidFontFamilyBranding',
      type: 'text'
    },
    {
      name: dark.typography.android.fontFamilyBrand,
      type: 'text'
    }
  ],
  [
    {
      name: 'iosFontFamilyBranding',
      type: 'text'
    },
    {
      name: dark.typography.iOS.fontFamilyBrand,
      type: 'text'
    }
  ]
);

const lightSchema = {
  head: ['name', 'color'],
  body: (light && light.palette) && Object.entries(light.palette).reduce(FlattenTheme, []).filter(removeStrings)
};

lightSchema.body.push(
  [
    {
      name: 'webFontFamilyBranding',
      type: 'text'
    },
    {
      name: light.typography.web.fontFamilyBrand,
      type: 'text'
    }
  ],
  [
    {
      name: 'androidFontFamilyBranding',
      type: 'text'
    },
    {
      name: light.typography.android.fontFamilyBrand,
      type: 'text'
    }
  ],
  [
    {
      name: 'iosFontFamilyBranding',
      type: 'text'
    },
    {
      name: light.typography.iOS.fontFamilyBrand,
      type: 'text'
    }
  ]
);

export const NaturaLightComponent = () => <Table schema={lightSchema} title="Light" />;
export const NaturaDarkComponent = () => <Table schema={darkSchema} title="Dark" />;
