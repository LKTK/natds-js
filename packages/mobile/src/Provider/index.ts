import { Provider as PaperProvider } from 'react-native-paper';

export class Provider extends PaperProvider{}
