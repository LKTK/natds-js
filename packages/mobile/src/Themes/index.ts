export { themes } from '@naturacosmeticos/natds-styles';
