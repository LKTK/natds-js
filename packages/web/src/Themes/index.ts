import {
  themes as styleThemes,
} from "@naturacosmeticos/natds-styles";
import { parseTheme } from "./parseTheme";
import { ThemeBrandName } from "./ThemeBrandName";
import createThemesObject from "./createThemesObject";
import { IThemesWeb } from "./IThemesWeb";

export { IThemeWeb } from "./IThemeWeb";

export const themes = Object.keys(styleThemes).reduce((value: IThemesWeb<ThemeBrandName>, key: string) => {

  const result = value;

  result[key as ThemeBrandName].dark = parseTheme(styleThemes[key as ThemeBrandName].dark);
  result[key as ThemeBrandName].light = parseTheme(styleThemes[key as ThemeBrandName].light);

  return result;
}, createThemesObject());
