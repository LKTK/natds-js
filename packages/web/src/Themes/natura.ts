import { themes, tokens } from '@naturacosmeticos/natds-styles';
import { createMuiTheme } from '@material-ui/core';

const { spacing }  = tokens.spacing;
const { naturaLight } = themes;
const { colors, roundness } = naturaLight;

const natdsTheme = createMuiTheme({
  spacing,
  palette: {
    primary: {
      main: colors.primary
    },
    secondary: {
      main: colors.secondary
    },
    error: {
      main: colors.error
    },
    type: 'light',
  },
  shape: {
    borderRadius: roundness
  }
});

export default natdsTheme;
