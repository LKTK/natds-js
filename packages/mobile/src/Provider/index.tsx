import { Provider as PaperProvider } from 'react-native-paper';
import * as React from 'react';
import { natura } from '../Themes';
import { ITheme } from '@naturacosmeticos/natds-styles';

interface IProvider {
  children: React.ReactNode;
  theme?: ITheme;
}

const Provider: React.FunctionComponent<IProvider> = (props) => {
  const { theme } = props;
  return <PaperProvider {...props} theme={theme ? theme : natura} />;
};

export default Provider;
