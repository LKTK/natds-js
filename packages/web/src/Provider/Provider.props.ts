import * as React from "react";
import { IThemeWeb } from "../Themes";

export interface IProviderProps {

  /**
   * @required
   * @type node
   */
  children: React.ReactNode;

  /**
   * Defaults to Natura Light theme
   *
   * @optional
   * @type IThemeWeb
   */
  theme?: IThemeWeb;
}
