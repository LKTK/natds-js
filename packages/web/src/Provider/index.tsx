/**
 * @todo Review/re-enable `jsx-no-string-ref` tslint rule
 * @todo Review/re-enable `no-unused-expression` tslint rule
 * @todo Review/re-enable `use-isnan` tslint rule
 * @todo Review/re-enable `triple-equals` tslint rule
 */
import { MuiThemeProvider } from '@material-ui/core/styles';
import createMuiTheme, { ThemeOptions } from '@material-ui/core/styles/createMuiTheme';
import * as React from 'react';

import { IThemeWeb, themes } from '../Themes';

export interface IProvider {
  children: React.ReactNode;
  theme?: IThemeWeb;
}

export const Provider:React.FunctionComponent<IProvider> = (props:IProvider) => {

  /**
   * @todo fix(web): TS2322 - Type is not assignable to type 'ThemeOptions'. Types of some properties are incompatible.
   */
  // @ts-ignore
  const theme: ThemeOptions = props.theme ? { ...props.theme } : { ...themes.natura.light },
    newTheme = createMuiTheme(theme);


  return (
    <MuiThemeProvider {...props} theme={newTheme} />
  );
};
