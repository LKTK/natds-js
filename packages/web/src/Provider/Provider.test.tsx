import * as React from "react";

import { createMuiTheme, ThemeOptions } from "@material-ui/core";
import { mount } from "enzyme";
import { themes } from "../Themes";
import { Provider } from "./Provider";

describe("Provider component", () => {
  it("should have the natura theme", () => {
    const provider = mount(<Provider>Mocked</Provider>);
    const muiTheme = createMuiTheme(themes.natura.light as unknown as ThemeOptions);
    const theme = provider.find("ThemeProvider").prop("theme");

    expect(JSON.stringify(theme)).toStrictEqual(JSON.stringify(muiTheme));
  });

  it("should have the provided theme", () => {
    const mockedTheme = themes.natura.dark;

    mockedTheme.palette = {
      ...mockedTheme.palette,
      primary: {
        main: "#000000",
      },
    };

    const provider = mount(<Provider theme={mockedTheme}>Mocked</Provider>);
    const theme = provider.find("ThemeProvider").prop("theme");

    expect(JSON.stringify(theme)).toStrictEqual(JSON.stringify(createMuiTheme(mockedTheme as unknown as ThemeOptions)));

  });
});
