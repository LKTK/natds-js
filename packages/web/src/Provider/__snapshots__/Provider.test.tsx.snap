// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Provider component when Natura light theme is provided should match snapshot 1`] = `"Provided content"`;

exports[`Provider component when customized Natura dark theme is provided should match snapshot 1`] = `"Provider content"`;

exports[`Provider component when no theme was provided should match snapshot 1`] = `"Provider content"`;
